### PR TITLE
Add undocumented but occasionally used attributes to inspect.pyi.

### PR DESCRIPTION
**Pull request by rchen152**
Additions based on:
https://github.com/python/cpython/blob/2.7/Lib/inspect.py
and
https://github.com/python/cpython/blob/3.6/Lib/inspect.py
Also fixes a couple of issues in stdlib/3/inspect.py: the signature of getframeinfo() was wrong, and Attribute was represented as a class when a typing.NamedTuple is the better choice.

### CODE REVIEW

File: stdlib/2/inspect.pyi [hunk 1]
```diff
@@ -1,7 +1,27 @@
-from types import TracebackType, FrameType, ModuleType
-from typing import Any, Dict, Callable, List, Optional, Tuple, Union, NamedTuple, Type
+from types import CodeType, TracebackType, FrameType, ModuleType
+from typing import Any, Dict, Callable, List, NamedTuple, Optional, Tuple, Type, Union
 
 # Types and members
+class EndOfBlock(Exception): ...
+
+class BlockFinder:
+  indent = ...  # type: int
```
Comment (JelleZijlstra) [inline stdlib/2/inspect.pyi]: Prefer PEP 526-style annotations (`indent: int`).

File: stdlib/2/inspect.pyi [hunk 2]
```diff
@@ -1,7 +1,27 @@
...
+  islambda = ...  # type: bool
+  started = ...  # type: bool
+  passline = ...  # type: bool
+  last = ...  # type: int
+  def tokeneater(self, type: int, token: str, srow_scol: Tuple[int, int],
```
Comment (JelleZijlstra) [inline stdlib/2/inspect.pyi]: Indent with four spaces.

File: stdlib/2/inspect.pyi [hunk 3]
```diff
@@ -32,6 +52,9 @@ def isgetsetdescriptor(object: object) -> bool: ...
 def ismemberdescriptor(object: object) -> bool: ...
 
 # Retrieving source code
+def findsource(object: object) -> Tuple[List[str], int]: ...
+def getabsfile(object: object) -> str: ...
+def getblock(lines: List[str]) -> List[str]: ...
```
Comment (JelleZijlstra) [inline stdlib/2/inspect.pyi]: I think this accepts an Iterable, not just a List.

File: stdlib/2/inspect.pyi [hunk 4]
```diff
@@ -61,6 +85,12 @@ ArgInfo = NamedTuple('ArgInfo', [('args', List[str]),
                                  ('locals', Dict[str, Any]),
                                  ])
 
+Arguments = NamedTuple('Arguments', [('args', List[Union[str, list]]),
```
Comment (JelleZijlstra) [inline stdlib/2/inspect.pyi]: Missing type arguments to the inner `list`.

Comment (rchen152) [reply stdlib/2/inspect.pyi]: The description in the code was "'args' is a list of argument names (possibly containing nested lists)", which I think means that the type is List[Union[str, List[Union[str, ...]]]].  Is there a way to type that precisely?

Comment (JelleZijlstra) [reply stdlib/2/inspect.pyi]: No, but in that case I'd prefer explicit `List[Any]` over `list`.

**Comment by rchen152** [comment]
Addressed all of the comments.

File: stdlib/2/inspect.pyi [hunk 5]
```diff
@@ -1,7 +1,27 @@
...
 # Types and members
+class EndOfBlock(Exception): ...
+
+class BlockFinder:
+    indent: int
+    islambda: bool
+    started: bool
+    passline: bool
+    last: int
+    def tokeneater(self, type: int, token: str, srow_scol: Tuple[int, int],
```
Comment (JelleZijlstra) [inline stdlib/2/inspect.pyi]: Why don't you include `indecorator`, `decoratorhasargs`, `last`?

Comment (rchen152) [reply stdlib/2/inspect.pyi]: I missed `indecorator` and `decoratorhasargs` because they're Python 3-specific - added now.  `last` is already present.

**Comment by ambv** [comment]
Thanks! ✨ 🍰 ✨

**Pull request merged**